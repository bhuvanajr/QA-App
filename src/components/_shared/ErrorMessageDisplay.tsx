import React from 'react';

type ErrorMessageDisplayProps = {
  id: string;
  message: string;
};

export const ErrorMessageDisplay: React.FunctionComponent<ErrorMessageDisplayProps> = props => {
  return (
    <div className="invalid-feedback d-block" data-testid="ErrorMessage" id={props.id + '_errormessage'}>
      <span className="icon icon-alert" />
      {' ' + props.message}
    </div>
  );
};
