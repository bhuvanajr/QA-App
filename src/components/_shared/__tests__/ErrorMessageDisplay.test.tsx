import { render } from '@testing-library/react';
import React from 'react';
import { ErrorMessageDisplay } from '../ErrorMessageDisplay';

test('should render ErrorMessageDisplay correctly', () => {
  const { container } = render(<ErrorMessageDisplay id="testId" message="Test error message" />);
  expect(container).toMatchSnapshot();
});
