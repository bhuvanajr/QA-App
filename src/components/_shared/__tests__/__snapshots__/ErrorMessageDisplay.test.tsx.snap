// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ErrorMessageDisplay correctly 1`] = `
<div>
  <div
    class="invalid-feedback d-block"
    data-testid="ErrorMessage"
    id="testId_errormessage"
  >
    <span
      class="icon icon-alert"
    />
     Test error message
  </div>
</div>
`;
